command/operator_client_state.go

```diff
@@ -0,0 +1,117 @@
+package command
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+	trstate "github.com/hashicorp/nomad/client/allocrunner/taskrunner/state"
+	"github.com/hashicorp/nomad/client/state"
+	"github.com/posener/complete"
+)
+
+type OperatorClientStateCommand struct {
+	Meta
+}
+
+func (c *OperatorClientStateCommand) Help() string {
+	helpText := `
+Usage: nomad operator client-state <path_to_nomad_dir>
+  Emits a json representation of the stored client state in json form.
+`
+	return strings.TrimSpace(helpText)
+}
+func (c *OperatorClientStateCommand) AutocompleteFlags() complete.Flags {
+	return complete.Flags{}
+}
+
+func (c *OperatorClientStateCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (c *OperatorClientStateCommand) Synopsis() string {
+	return "Dump the nomad client state"
+}
+func (c *OperatorClientStateCommand) Name() string { return "operator client-state" }
+
+func (c *OperatorClientStateCommand) Run(args []string) int {
+	if len(args) != 1 {
+		c.Ui.Error("This command takes one argument: <nomad-data-dir>")
+		c.Ui.Error(commandErrorText(c))
+
+		return 1
+	}
+
+	logger := hclog.L()
+	db, err := state.NewBoltStateDB(logger, args[0])
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("failed to open client state: %v\n", err))
```

Comment:
`c.Ui.Error` and `c.Ui.Output` do the right thing here around trailing newlines, so we don't need this bit:

```suggestion
		c.Ui.Error(fmt.Sprintf("failed to open client state: %v", err))
```